src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Create/CreateResourceHandler.cs

```diff
@@ -59,7 +59,7 @@ public async Task<UpsertResourceResponse> Handle(CreateResourceRequest request,
 
             await _referenceResolver.ResolveReferencesAsync(resource, _referenceIdDictionary, resource.TypeName, cancellationToken);
 
-            ResourceWrapper resourceWrapper = CreateResourceWrapper(resource, deleted: false, keepMeta: true);
+            ResourceWrapper resourceWrapper = ResourceWrapperFactory.CreateResourceWrapper(resource, ResourceIdProvider, deleted: false, keepMeta: true);
```

Comment:
This looks like it should be an extension method

Reply:
It is, it is part of ResourceWrapperFactoryExtensions.